File: src/main/java/io/nightfis/createstellarexoduscore/filter/IsPetFilter.java
package io.nightfis.createstellarexoduscore.filter;

import net.minecraft.network.chat.Component;
import net.minecraft.world.entity.LivingEntity;
import net.minecraft.world.entity.TamableAnimal;

public class IsPetFilter extends TargetFilter {

    private static final String ID = "IsPetFilter";

    @Override
    public String getId() {
        return ID;
    }

    @Override
    public boolean match(LivingEntity entity, String arg) {
        return entity instanceof TamableAnimal tamableAnimal && tamableAnimal.getOwner() != null;
    }

    @Override
    public Component title() {
        return Component.translatable("screen.createstellarexoduscore.auto_aim_turret.filter.is_pet");
    }
}

File: src/main/kotlin/io/nightfis/createstellarexoduscore/filter/IsPetFilter.kt
package io.nightfis.createstellarexoduscore.filter

import net.minecraft.network.chat.Component
import net.minecraft.world.entity.LivingEntity
import net.minecraft.world.entity.TamableAnimal

class IsPetFilter : TargetFilter() {

    override fun getId(): String = ID

    override fun match(entity: LivingEntity, arg: String): Boolean {
        return entity is TamableAnimal && entity.owner != null
    }

    override fun title(): Component {
        return Component.translatable("screen.create_stellar_exodus_core.auto_aim_turret.filter.is_pet")
    }

    companion object {
        private const val ID = "IsPetFilter"
    }
}
